app/controllers/partner_users_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_String_literal: true
+
+class PartnerUsersController < ApplicationController
+  before_action :set_partner, only: %i[index create destroy resend_invitation]
+
+  def index
+    @users = @partner.users
+    @user = User.new(name: "")
+  end
+
+  def create
+    @user = UserInviteService.invite(
+      email: user_params[:email],
+      name: user_params[:name],
+      roles: [Role::PARTNER],
+      resource: @partner
+    )
+    if @user.valid?
+      redirect_back(fallback_location: "/",
+        notice: "#{@user.name} has been invited. Invitation email sent to #{@user.email}")
+    else
+      flash[:alert] = "Invitation failed. Check the form for errors."
+      @users = @partner.users
+      render :index
+    end
+  end
+
+  def destroy
+    user = User.find(params[:id])
+
+    if user.remove_role(Role::PARTNER, @partner)
+      redirect_back(fallback_location: "/", notice: "Access to #{user.name} has been revoked.")
+    else
+      redirect_back(fallback_location: "/", alert: "Invitation failed. Check the form for errors.")
+    end
+  end
+
+  def resend_invitation
+    user = User.find(params[:id])
+
+    if user.invitation_accepted_at.nil?
+      user.invite!
+    else
+      user.errors.add(:base, "User has already accepted invitation.")
+    end
```

Comment:
Can replace this with `UserInviteService` just like above, just pass `force: true`.

Reply:
Yep I think so!

Reply:
I'am actually thinking that a new method makes sense, maybe `self.resend_invite`? It feels like that method is getting more complex for something that we just want to resend the invite (not worry about Roles). What do you think?

Reply:
That's fair - we can add a separate method to the service that only works on an existing user and doesn't change the roles.